File: PAP_PHP/VERSÂO FINAL FIRE/app/models/Comentario.php
<?php

class Comentario {
    private int $id;
    private int $id_post;
    private int $id_utilizador;
    private ?int $id_comentario_pai;
    private string $dt_comentario;
    private string $texto_comentario;
    private int $like_count;
    private int $comment_count;
    private ?string $username;
    private ?string $image;

    public function __construct(
        int $id = 0,
        int $id_post = 0,
        int $id_utilizador = 0,
        ?int $id_comentario_pai = null,
        string $dt_comentario = '',
        string $texto_comentario = '',
        int $like_count = 0,
        int $comment_count = 0,
        ?string $username = null,
        ?string $image = null
    ) {
        $this->id                = $id;
        $this->id_post           = $id_post;
        $this->id_utilizador     = $id_utilizador;
        $this->id_comentario_pai = $id_comentario_pai;
        $this->dt_comentario     = $dt_comentario;
        $this->texto_comentario  = $texto_comentario;
        $this->like_count        = $like_count;
        $this->comment_count     = $comment_count;
        $this->username          = $username;
        $this->image             = $image;
    }

    public function getId(): int { return $this->id; }
    public function setId(int $id): void { $this->id = $id; }

    public function getIdPost(): int { return $this->id_post; }
    public function setIdPost(int $id_post): void { $this->id_post = $id_post; }

    public function getIdUtilizador(): int { return $this->id_utilizador; }
    public function setIdUtilizador(int $id_utilizador): void { $this->id_utilizador = $id_utilizador; }

    public function getIdComentarioPai(): ?int { return $this->id_comentario_pai; }
    public function setIdComentarioPai(?int $id_comentario_pai): void { $this->id_comentario_pai = $id_comentario_pai; }

    public function getDtComentario(): string { return $this->dt_comentario; }
    public function setDtComentario(string $dt_comentario): void { $this->dt_comentario = $dt_comentario; }

    public function getTextoComentario(): string { return $this->texto_comentario; }
    public function setTextoComentario(string $texto_comentario): void { $this->texto_comentario = $texto_comentario; }

    public function getLikeCount(): int { return $this->like_count; }
    public function setLikeCount(int $like_count): void { $this->like_count = $like_count; }

    public function getCommentCount(): int { return $this->comment_count; }
    public function setCommentCount(int $comment_count): void { $this->comment_count = $comment_count; }

    public function getUsername(): ?string { return $this->username; }
    public function getImage(): ?string { return $this->image; }

    public function toArray(): array {
        return [
            'id'                => $this->id,
            'id_post'           => $this->id_post,
            'id_utilizador'     => $this->id_utilizador,
            'id_comentario_pai' => $this->id_comentario_pai,
            'dt_comentario'     => $this->dt_comentario,
            'texto_comentario'  => $this->texto_comentario,
            'like_count'        => $this->like_count,
            'comment_count'     => $this->comment_count,
            'username'          => $this->username,
            'image'             => $this->image
        ];
    }
}

File: PAP_APP/Imoral/app/src/main/java/com/example/imoral/models/Comentario.java
package com.example.imoral.models;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

public class Comentario implements Serializable {
    private int id;
    private int id_post;
    private int id_utilizador;
    private Integer id_comentario_pai;
    private String dt_comentario;
    private String texto_comentario;
    private int like_count;
    private int comment_count;
    private String username;
    private String image;
    private boolean mostrarRespostas = false;
    private List<Comentario> respostas = new ArrayList<>();

    public Comentario() {}

    public int getId() { return id; }
    public void setId(int id) { this.id = id; }

    public int getId_post() { return id_post; }
    public void setId_post(int id_post) { this.id_post = id_post; }

    public int getId_utilizador() { return id_utilizador; }
    public void setId_utilizador(int id_utilizador) { this.id_utilizador = id_utilizador; }

    public Integer getId_comentario_pai() { return id_comentario_pai; }
    public void setId_comentario_pai(Integer id_comentario_pai) { this.id_comentario_pai = id_comentario_pai; }

    public String getDt_comentario() { return dt_comentario; }
    public void setDt_comentario(String dt_comentario) { this.dt_comentario = dt_comentario; }

    public String getTexto_comentario() { return texto_comentario; }
    public void setTexto_comentario(String texto_comentario) { this.texto_comentario = texto_comentario; }

    public int getLike_count() { return like_count; }
    public void setLike_count(int like_count) { this.like_count = like_count; }

    public int getComment_count() { return comment_count; }
    public void setComment_count(int comment_count) { this.comment_count = comment_count; }

    public String getUsername() { return username; }
    public void setUsername(String username) { this.username = username; }

    public String getImage() { return image; }
    public void setImage(String image) { this.image = image; }

    public boolean isMostrarRespostas() { return mostrarRespostas; }
    public void setMostrarRespostas(boolean mostrarRespostas) { this.mostrarRespostas = mostrarRespostas; }

    public List<Comentario> getRespostas() { return respostas; }
    public void setRespostas(List<Comentario> respostas) { this.respostas = respostas; }
}



